addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homeassistant/MappingJsonReader.java

```diff
@@ -0,0 +1,182 @@
+        ABBREVIATIONS.put("sup_feat", "supported_features");
+        ABBREVIATIONS.put("swing_mode_cmd_t", "swing_mode_command_topic");
+        ABBREVIATIONS.put("swing_mode_stat_tpl", "swing_mode_state_template");
+        ABBREVIATIONS.put("swing_mode_stat_t", "swing_mode_state_topic");
+        ABBREVIATIONS.put("temp_cmd_t", "temperature_command_topic");
+        ABBREVIATIONS.put("temp_stat_tpl", "temperature_state_template");
+        ABBREVIATIONS.put("temp_stat_t", "temperature_state_topic");
+        ABBREVIATIONS.put("tilt_clsd_val", "tilt_closed_value");
+        ABBREVIATIONS.put("tilt_cmd_t", "tilt_command_topic");
+        ABBREVIATIONS.put("tilt_inv_stat", "tilt_invert_state");
+        ABBREVIATIONS.put("tilt_max", "tilt_max");
+        ABBREVIATIONS.put("tilt_min", "tilt_min");
+        ABBREVIATIONS.put("tilt_opnd_val", "tilt_opened_value");
+        ABBREVIATIONS.put("tilt_status_opt", "tilt_status_optimistic");
+        ABBREVIATIONS.put("tilt_status_t", "tilt_status_topic");
+        ABBREVIATIONS.put("t", "topic");
+        ABBREVIATIONS.put("uniq_id", "unique_id");
+        ABBREVIATIONS.put("unit_of_meas", "unit_of_measurement");
+        ABBREVIATIONS.put("val_tpl", "value_template");
+        ABBREVIATIONS.put("whit_val_cmd_t", "white_value_command_topic");
+        ABBREVIATIONS.put("whit_val_stat_t", "white_value_state_topic");
+        ABBREVIATIONS.put("whit_val_tpl", "white_value_template");
+        ABBREVIATIONS.put("xy_cmd_t", "xy_command_topic");
+        ABBREVIATIONS.put("xy_stat_t", "xy_state_topic");
+        ABBREVIATIONS.put("xy_val_tpl", "xy_value_template");
+
+        DEVICE_ABBREVIATIONS.put("cns", "connections");
+        DEVICE_ABBREVIATIONS.put("ids", "identifiers");
+        DEVICE_ABBREVIATIONS.put("name", "name");
+        DEVICE_ABBREVIATIONS.put("mf", "manufacturer");
+        DEVICE_ABBREVIATIONS.put("mdl", "model");
+        DEVICE_ABBREVIATIONS.put("sw", "sw_version");
+    }
+
+    private final Map<String, String> mapping;
+
+    /**
+     *
+     * @param delegate
+     * @return return a JsonReader which replaces all config abbreviations
+     */
+    public static MappingJsonReader getConfigMapper(JsonReader delegate) {
+        return new MappingJsonReader(JsonReaderDelegate.getDelegate(delegate), ABBREVIATIONS);
+    }
+
+    /**
+     *
+     * @param delegate
+     * @return return a JsonReader which replaces all config.device abbreviations
+     */
+    public static MappingJsonReader getDeviceMapper(JsonReader delegate) {
+        return new MappingJsonReader(JsonReaderDelegate.getDelegate(delegate), DEVICE_ABBREVIATIONS);
+    }
+
+    private MappingJsonReader(JsonReader delegate, Map<String, String> mapping) {
+        super(delegate);
+        this.mapping = mapping;
+    }
+
+    @Override
+    public String nextName() throws IOException {
+        String name = super.nextName();
+        return mapping.getOrDefault(name, name);
+    }
+
+}
```

Comment:
new line

Reply:
The EHS Formatter options don't seem to have that set....
I now have it set (New Lines->At end of file)